### ISSUE TITLE
Unrecognised JIAWEN bulbs

### ISSUE THREAD

**Issue by AntiHeld889** [ebaauw/homebridge-hue#171]
Hey Erik,
When you have a chance, would you mind adding the following lights to the "known light model" list?

Sep 26 18:35:45 raspberrypi homebridge[26717]: [2017-9-26 18:35:45] [RaspBee] deCONZ-GW: /lights/21: warning: unknown light model {"etag":"752659775e2fef90ab325b7a93de5b85","hascolor":true,"manufacturername":"Unknown","modelid":"FB56-ZCW08KU1.1","name":"China","state":{"alert":"none","bri":255,"colormode":"xy","ct":495,"effect":"none","hue":14964,"on":false,"reachable":true,"sat":8,"xy":[0.293586,0.644097]},"swversion":"1.0.2","type":"Extended color light","uniqueid":"00:12:4b:00:11:e8:27:38-0b"}

**Comment by ebaauw** [comment]
Doesn't the light report a _Manufacturer Name_?  What is shown in the deCONZ GUI when you read the attributes of the Basic cluster in the Cluster Info panel?

Could you please install [`ph.sh`](https://github.com/ebaauw/ph.sh) and run the following from the command line?
```bash
ph_verbose=true ph_light_values 21
```
This should determine the colour temperature and colour range this light supports.  You can speed up the discovery by reading the attributes for the Color Control cluster in the deCONZ GUI while `ph_light_values` is waiting for the gateway to poll the light.

**Comment by AntiHeld889** [comment]
After reading the Basic Cluster:

[RaspBee] deCONZ-GW: /lights/21: warning: unknown light model {"etag":"5283c684ea242ff3375aada12e112eb1","hascolor":true,"manufacturername":"Feibit Inc co.","modelid":"FB56-ZCW08KU1.1","name":"China","state":{"alert":"none","bri":254,"colormode":"ct","ct":450,"effect":"none","hue":4644,"on":false,"reachable":true,"sat":108,"xy":[0.145085,0.224498]},"swversion":"1.0.2","type":"Extended color light","uniqueid":"00:12:4b:00:11:e8:27:38-0b"}

But if I run "ph_verbose=true ph_light_values 21" it shows me:

pi@raspberrypi:~ $ ph_verbose=true ph_light_values 21
localhost: /lights/21: Feibit Inc co. FB56-ZCW08KU1.1 Extended color light "China"
localhost: /lights/21: ct: cool ............................................................... 153
localhost: /lights/21: ct: warm ............................................................... 500
localhost: /lights/21: xy: red ...........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

The light bulb changes from white to green, but nothing else happens. I have waited 30 min but there will be no change.

**Comment by ebaauw** [comment]
Ok, thanks.  I’ll add the light with the default gamut for now.

I’ll try and include the probing in homebridge-hue - the script seems to cause too many issues.

**Comment by ebaauw** [comment]
In v0.5.33.

**Comment by ebaauw** [comment]
On second thought, this might not be an issue with `ph_light_values` but with the light itself.  I think it echoes the values set, like the IKEA colour bulb, rather than the actual colour, like the Hue or innr colour bulbs (see https://github.com/dresden-elektronik/deconz-rest-plugin/issues/218).  Also, I would expect the light to change to red.  That it changes to green instead is probably because it doesn't like the value 0 for _CurrentY_.

Note to self: homebridge-hue should minimise the value for y to 0.0001.

**Comment by ebaauw** [comment]
@AntiHeld889 , may I ask you to try again, using the javascript version of `ph`, which is included in homebridge-hue:
```
ph probe -v /lights/21
```

It should produce a result like:
```
ph probe: /lights/221: red .... done
ph probe: /lights/221: green ... done
ph probe: /lights/221: blue ... done
{
  "manufacturername": "Philips",
  "modelid": "LCT003",
  "type": "Extended color light",
  "bri": true,
  "ct": {
    "min": 153,
    "max": 500
  },
  "xy": {
    "r": [
      0.675,
      0.322
    ],
    "g": [
      0.409,
      0.518
    ],
    "b": [
      0.167,
      0.04
    ]
  }
}
```
The Philips light reports back the actual colour, and deCONZ now fast-polls the light after setting it, hence you don't see any wait times.  For lights that don't report the actual colour, it will look like:
```
ph probe: /lights/7: red ............................................................... timeout
ph probe: /lights/7: green ............................................................... timeout
ph probe: /lights/7: blue ............................................................... timeout
{
  "manufacturername": "IKEA of Sweden",
  "modelid": "TRADFRI bulb E27 CWS opal 600lm",
  "type": "Color light",
  "bri": true,
  "xy": {
    "r": [
      0.9961,
      0.0001
    ],
    "g": [
      0.0001,
      0.9961
    ],
    "b": [
      0.0001,
      0.0001
    ]
  }
}
```

**Comment by AntiHeld889** [comment]
hi Erik, here is my result

probe -v /lights/9
ph probe: /lights/9: cool ............................................................... timeout
ph probe: /lights/9: warm ............................................................... timeout
ph probe: /lights/9: red ... done
ph probe: /lights/9: green ... done
ph probe: /lights/9: blue ... done
{
  "manufacturername": "Feibit Inc co.",
  "modelid": "FB56-ZCW08KU1.1",
  "type": "Extended color light",
  "bri": true,
  "ct": {
    "min": 1,
    "max": 1000
  },
  "xy": {
    "r": [
      0.996094,
      0.0000919132
    ],
    "g": [
      0.0000919132,
      0.996094
    ],
    "b": [
      0.0000919132,
      0.0000919132
    ]
  }
}

**Comment by ebaauw** [comment]
Thanks, @AntiHeld889

Darn, there's still an issue with `ph probe`: it doesn't realise it's connected to deCONZ.  I fixed this in homebridge-hue-utils v0.4.4.

I'm really sorry to keep bothering you, but when you're up to it, could you try again?

You can upgrade homebridge-hue-utils by re-installing the _currently installed version_ of homebridge-hue, see below.  Be careful not to install homebride-hue v0.8.x before you're ready - it has breaking changes, see https://github.com/ebaauw/homebridge-hue/releases.  There is no hurry - you can also try again in a couple of weeks/months, after you've upgraded homebridge-hue.

```
$ ph -V
0.4.3
$ sudo npm -g list --depth=0
/usr/lib
├── homebridge@0.4.38
├── homebridge-hue@0.7.1
├── homebridge-p1@0.0.2
└── npm@5.6.0

$ sudo npm -g i homebridge-hue@0.7.1
/usr/bin/dc_eventlog -> /usr/lib/node_modules/homebridge-hue/node_modules/homebridge-hue-utils/cli/dc_eventlog.js
/usr/bin/json -> /usr/lib/node_modules/homebridge-hue/node_modules/homebridge-hue-utils/cli/json.js
/usr/bin/ph -> /usr/lib/node_modules/homebridge-hue/node_modules/homebridge-hue-utils/cli/ph.js
/usr/bin/upnp -> /usr/lib/node_modules/homebridge-hue/node_modules/homebridge-hue-utils/cli/upnp.js
+ homebridge-hue@0.7.1
updated 9 packages in 14.26s


   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.6.0 → 5.7.1    │
   │     Run npm i -g npm to update      │
   │                                     │
   ╰─────────────────────────────────────╯

$ ph -V
0.4.4
$ sudo npm -g list --depth=0
/usr/lib
├── homebridge@0.4.38
├── homebridge-hue@0.7.1
├── homebridge-p1@0.0.2
└── npm@5.6.0

```
